test/unit/puppetlabs/services/jruby/jruby_puppet_testutils.clj

```diff
@@ -181,3 +189,114 @@ create-mock-pool-instance :- JRubyInstance
     (while (< (count (jruby-core/registered-instances pool-context))
               num-jrubies)
       (Thread/sleep 100))))
+
+(schema/defn ^:always-validate
+  mock-puppet-config-settings :- {schema/Str schema/Any}
+  "Return a map of settings that mock the settings that core Ruby Puppet
+  would return via a call to JRubyPuppet.getSetting()."
+  [jruby-puppet-config :- {:master-conf-dir schema/Str
+                           :master-code-dir schema/Str
+                           schema/Keyword schema/Any}]
+  (let [certname "localhost"
+        confdir (:master-conf-dir jruby-puppet-config)
+        ssldir (str confdir "/ssl")
+        certdir (str ssldir "/certs")
+        cadir (str ssldir "/ca")
+        private-key-dir (str ssldir "/private_keys")]
+    {"allow_duplicate_certs" false
+     "autosign" true
+     "keylength" 2048
+     "cacert" (str cadir "/ca_crt.pem")
+     "ca_name" (str "Puppet CA: " certname)
+     "cacrl" (str cadir "/ca_crl.pem")
+     "cakey" (str cadir "/ca_key.pem")
+     "capub" (str cadir "/ca_pub.pem")
+     "ca_ttl" 157680000
+     "certdir" certdir
+     "certname" certname
+     "cert_inventory" (str cadir "/inventory.txt")
+     "codedir" (:master-code-dir jruby-puppet-config)
+     "csr_attributes" (str confdir "/csr_attributes.yaml")
+     "csrdir" (str cadir "/requests")
+     "data_binding_terminus" "hiera"
+     "dns_alt_names" ""
+     "hostcert" (str certdir "/" certname ".pem")
+     "hostcrl" (str ssldir "/crl.pem")
+     "hostprivkey" (str private-key-dir "/" certname ".pem")
+     "hostpubkey" (str ssldir "/public_keys/" certname ".pem")
+     "localcacert" (str certdir "/ca.pem")
+     "manage_internal_file_permissions" true
+     "privatekeydir" private-key-dir
+     "requestdir" (str ssldir "/certificate_requests")
+     "serial" (str cadir "/serial")
+     "signeddir" (str cadir "/signed")
+     "ssl_client_header" "HTTP_X_CLIENT_DN"
+     "ssl_client_verify_header" "HTTP_X_CLIENT_VERIFY"
+     "trusted_oid_mapping_file" (str confdir
+                                     "/custom_trusted_oid_mapping.yaml")}))
+
+(schema/defn ^:always-validate
+  create-mock-jruby-puppet :- JRubyPuppet
+  "Create a 'mock' JRubyPuppet instance which returns fixed values for settings
+  and puppet version and a hard-coded HTTP 200 response for any requests it
+  handles."
+  [config :- {schema/Keyword schema/Any}]
+  (let [puppet-config (merge
+                       (mock-puppet-config-settings (:jruby-puppet config))
+                       (:puppet config))]
+    (reify JRubyPuppet
+      (getSetting [_ setting]
+        (get puppet-config setting))
+      (puppetVersion [_]
+        "1.2.3")
+      (handleRequest [_ _]
+        (JRubyPuppetResponse. (Integer. 200)
+                              "looks good to me"
+                              "text/plain"
+                              "1.2.3"))
```

Comment:
Might be worth talking this one through a little more once we get closer to a final version of this PR.  My gut is that any test that is actually making a call down into `handleRequest` should probably not be getting mocked, and if there are only a few tests that are relying on this, it might be a worthwhile tradeoff in my mind to just let those tests be a little slower for now.  But it might also depend on which tests we're talking about... we can circle back around to this.